test/unit/runner.spec.js

```diff
@@ -487,6 +487,32 @@ describe('Runner', function() {
         });
         runner.failHook(hook, err);
       });
+
+      it('should not hang if the error message is ridiculously long', function(done) {
+        var hook = new Hook();
+        var message = [];
+        // mock a long message
+        for (var i = 0; i < 20000; i++) {
+          var line = stack[i % stack.length];
+          message[i] = line[Math.floor(Math.random() * line.length)];
+        }
+        var err = new Error(message);
+        // Fake stack-trace
+        err.stack = [message].concat(stack).join('\n');
+
+        runner.on('fail', function(hook, err) {
+          expect(
+            err.stack
+              .split('\n')
+              .slice(1)
+              .join('\n'),
+            'to be',
+            stack.slice(0, 3).join('\n')
+          );
+          done();
+        });
+        runner.failHook(hook, err);
+      });
```

Comment:
`./bin/mocha test/unit/runner.spec.js --grep longStackTrace --timeout 1000` will hang and then timeout if regex were reverted.